### ISSUE TITLE
SAML authentication request has empty AllowCreate attribute

### ISSUE THREAD

**Issue by rizz360** [directus/directus#21995]
### Describe the Bug

As of 10.10+ AllowCreate appears to be empty in the SAML auth request:

## 10.9 
```xml
<samlp:AuthnRequest ...>
 <saml:Issuer>...</saml:Issuer>
 <samlp:NameIDPolicy ... AllowCreate="false"></samlp:NameIDPolicy>
</samlp:AuthnRequest>
```

## 10.10 
```xml
<samlp:AuthnRequest ...>
 <saml:Issuer>...</saml:Issuer>
 <samlp:NameIDPolicy ... AllowCreate=""></samlp:NameIDPolicy>
</samlp:AuthnRequest>
```

With my local testing setup this leads to the following error with SimpleSAML:

```
SimpleSAML_Error_Error: UNHANDLEDEXCEPTION

Backtrace:
1 www/_include.php:45 (SimpleSAML_exception_handler)
0 [builtin] (N/A)
Caused by: Exception: Invalid value of boolean attribute 'AllowCreate': ''
...
```

`AUTH_xxxxxx_ALLOW_PUBLIC_REGISTRATION` does not seem to have any effect on this.

I retried the setup in a new project with the same result.

### To Reproduce

- Default Docker compose setup with SAML authentication set up (see below)
- Triggering saml authentication request throws an error by the IDP since AllowCreate is empty
- Switching back to 10.9.x makes it work again since AllowCreate="false" in the request
<details>
  <summary>Docker compose - Click to expand</summary>

```yaml
version: "3"
volumes:
  directus_data:
services:
  database: ...
  cache: ...
  directus:
    container_name: directus
    image: directus/directus:10.10.4
    ports:
      - ${DIRECTUS_PORT}:8055
    volumes: ...
    depends_on: ...
    healthcheck: ...
    environment:
      KEY: ${DIRECTUS_KEY}
      SECRET: ${DIRECTUS_SECRET}
      ...
      # Make sure to set this in production
      # (see https://docs.directus.io/self-hosted/config-options#general)
      PUBLIC_URL: "http://localhost:${DIRECTUS_PORT}"
     ...
      CORS_ENABLED: "true"
      CORS_ORIGIN: "true"

      # SAML RELATED ENV VARIABLES
      AUTH_PROVIDERS: "IAM"
      AUTH_IAM_DRIVER: "saml"
      AUTH_IAM_IDP_metadata: |
        <md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" entityID="http://localhost:8880/simplesaml/saml2/idp/metadata.php">
          <md:IDPSSODescriptor protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
            <md:KeyDescriptor use="signing">
              <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
                <ds:X509Data>
                  <ds:X509Certificate>...</ds:X509Certificate>
                </ds:X509Data>
              </ds:KeyInfo>
            </md:KeyDescriptor>
            <md:KeyDescriptor use="encryption">
              <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
                <ds:X509Data>
                  <ds:X509Certificate>...</ds:X509Certificate>
                </ds:X509Data>
              </ds:KeyInfo>
            </md:KeyDescriptor>
            <md:SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="http://localhost:8880/simplesaml/saml2/idp/SingleLogoutService.php"/>
            <md:NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:transient</md:NameIDFormat>
            <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="http://localhost:8880/simplesaml/saml2/idp/SSOService.php"/>
          </md:IDPSSODescriptor>
        </md:EntityDescriptor>

      AUTH_IAM_SP_metadata: |
        <EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" entityID="[some id]">
          <SPSSODescriptor AuthnRequestsSigned="false" WantAssertionsSigned="true" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
              <NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:transient</NameIDFormat>
              <!-- NameIDPolicy -->
              <NameIDPolicy Format="urn:oasis:names:tc:SAML:2.0:nameid-format:transient"/>
              <!-- Assertion Consumer Service URL -->
              <AssertionConsumerService isDefault="true" Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="http://localhost:${DIRECTUS_PORT}/auth/login/IAM/acs" index="0"/>
              <!-- Logout URL -->
              <SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="http://localhost:${DIRECTUS_PORT}/auth/logout"/>
          </SPSSODescriptor>
        </EntityDescriptor>
      AUTH_IAM_ALLOW_PUBLIC_REGISTRATION: true
      AUTH_IAM_DEFAULT_ROLE_ID: "ac61c47f-4c0f-47b8-84fd-945ed7e5e0a8" # user role
      AUTH_IAM_IDENTIFIER_KEY: "urn:oid:0.9.2342.19200300.100.1.1"
      AUTH_IAM_EMAIL_KEY: "urn:oid:0.9.2342.19200300.100.1.3"
      AUTH_IAM_ICON: "verified_user"
      AUTH_IAM_LABEL: "IAM"
      AUTH_IAM_LOGOUT_URL: "http://localhost:8880/simplesaml/saml2/idp/SingleLogoutService.php"
      AUTH_IAM_REDIRECT_ALLOW_LIST: http://localhost:${DIRECTUS_PORT}/admin/login,http://localhost:8055/admin/login?continue=, http://localhost:4200/home, http://localhost:4200/
      ...
  # Only used for debugging SAML authentication
  auth-saml:
    container_name: auth-saml
    image: kristophjunge/test-saml-idp
    ports:
      - 8880:8080
    environment:
      - SIMPLESAMLPHP_SP_ENTITY_ID=[some id]
      - SIMPLESAMLPHP_SP_ASSERTION_CONSUMER_SERVICE=${SP_ACS}
    extra_hosts:
      - "host.docker.internal:host-gateway"
    volumes:
      - ./auth-saml-config/users.php:/var/www/simplesamlphp/config/authsources.php
    healthcheck:
      test:
        [
          "CMD",
          "wget",
          "--no-verbose",
          "--tries=1",
          "--spider",
          "http://localhost:8080/",
          "||",
          "exit",
          "1",
        ]
      interval: 1m
      timeout: 10s
      retries: 3
```
</details>

### Directus Version

10.10.4

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by rizz360** [comment]
Hi there,

Hope you're all doing well! It's been about a month since the last update on this issue. Any progress or insights to share? If you need any more information from my end, feel free to ask.

Thanks a lot!

**Comment by paescuj** [comment]
Thanks for your patient, this will be addressed in #22340.

In the meantime you can use the following workaround:
```yaml
    environment:
      # ...
      AUTH_IAM_SP_allowCreate: "string:false"
```

**Comment by rizz360** [comment]
That's great, thank you! I will give it a try.
